### DISCUSSION TITLE
Problems with `count`with a `group_by` clause on multiple columns

### DISCUSSION THREAD

**Discussion by SuperPommeDeTerre** [diesel-rs/diesel#3682] [Q&A] [answered]
Hi everyone,

I have the following code that is somewhat simple, but I can't figure out to make it works (and compile):
```rust
       let conn = &mut db::connection()?;
        let devisdb_query = devis::table
            .left_join(imputations::table)
            .into_boxed()
            .group_by(devis::all_columns)
            .select((
                devis::all_columns,
                count(imputations::id) / 2,
            ))
            .filter(devis::id.eq(id));
        let devisdb = devisdb_query.first::<QuotationReturnedToClient>(conn)?;
```

I have some compile errors that I don't understand:
```
error[E0271]: type mismatch resolving `<table as AppearsInFromClause<table>>::Count == Never`
   --> src/models/devis.rs:85:14
    |
85  |             .select((
    |              ^^^^^^ expected `Once`, found `Never`
```
Can someone help me understand wht's going on ?

I'm using diesel v2.1.0.

**Comment by weiznich** [comment, marked as answer]
> `select` has slightly stricter bounds on its arguments than other methods. In particular, when used with a left outer join, `.nullable` must be called on columns that come from the right side of a join. It can be called on the column itself, or on an expression containing that column. `title.nullable()`, `lower(title).nullable()`, and `(id, title).nullable()` would all be valid.

[Source](https://docs.diesel.rs/2.1.x/diesel/prelude/trait.QueryDsl.html#method.select)
